src/modules/useFlats.ts

```diff
@@ -1,18 +1,26 @@
 import {useState} from "react";
 import FlatService from "services/FlatService";
 import {Flat} from "common/types/Flat";
+import {useDispatch} from "react-redux";
+import {setPagination} from "../store/modules/pagination/actions";
+import {FLATS, FLATS_MAX} from "../store/modules/pagination/types";
 
 const useFlats = () => {
     const [flatsLoading, setFlatsLoading] = useState(false);
     const [flats, setFlats] = useState<Flat[]>([]);
     const [flat, setFlat] = useState<Flat>();
+    const dispatch = useDispatch();
 
-    const fetchFlats = () => {
+    const fetchFlats = (page: number | null = null) => {
         return new Promise((resolve, reject) => {
             setFlatsLoading(true);
-            return FlatService.index()
+            return FlatService.index(page === null ? null : {page: page})
```

Comment:
jesli w obiekcie przekazesz { page } to on automatycznie zamienia to na { page: page } wiec wystarczy:
page=== null  ? null : { page }